core/src/main/scala/kafka/log/AbstractIndex.scala

```diff
@@ -114,6 +116,9 @@ abstract class AbstractIndex[K, V](@volatile var file: File, val baseOffset: Lon
         raf.setLength(roundedNewSize)
         mmap = raf.getChannel().map(FileChannel.MapMode.READ_WRITE, 0, roundedNewSize)
         _maxEntries = mmap.limit / entrySize
+        if(newlyCreated){
+          mmap.put(contentBuffer.asInstanceOf[ByteBuffer])
+        }
         mmap.position(position)
```

Comment:
Since ByteBuffer#put increments the position, do we have to set the position again here? 

Besides, seems `contentBuffer` will not be used anymore after  `resize` completes, is it a good idea to explicitly unmap this buffer like what we did in [KAFKA-4614](https://issues.apache.org/jira/browse/KAFKA-4614)?